pkg/scanners/trivy/trivy.go

```diff
@@ -0,0 +1,435 @@
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	"github.com/Azure/eraser/pkg/logger"
+	"github.com/aquasecurity/fanal/analyzer/config"
+	"github.com/aquasecurity/fanal/applier"
+	"github.com/aquasecurity/fanal/artifact"
+	artifactImage "github.com/aquasecurity/fanal/artifact/image"
+	"github.com/aquasecurity/fanal/cache"
+	fanalImage "github.com/aquasecurity/fanal/image"
+	fanalTypes "github.com/aquasecurity/fanal/types"
+	"github.com/aquasecurity/trivy-db/pkg/db"
+	dlDb "github.com/aquasecurity/trivy/pkg/db"
+	"github.com/aquasecurity/trivy/pkg/detector/ospkg"
+	pkgResult "github.com/aquasecurity/trivy/pkg/result"
+	"github.com/aquasecurity/trivy/pkg/scanner"
+	"github.com/aquasecurity/trivy/pkg/scanner/local"
+	trivyTypes "github.com/aquasecurity/trivy/pkg/types"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	generalErr = 1
+
+	severityCritical = "CRITICAL"
+	severityHigh     = "HIGH"
+	severityMedium   = "MEDIUM"
+	severityLow      = "LOW"
+	severityUnknown  = "UNKNOWN"
+
+	vulnTypeOs      = "os"
+	vulnTypeLibrary = "library"
+
+	securityCheckVuln   = "vuln"
+	securityCheckConfig = "config"
+	securityCheckSecret = "secret"
+
+	apiPath         = "apis/eraser.sh/v1alpha1"
+	resourceName    = "imagecollectors"
+	subResourceName = "status"
+	trivyVersion    = "v0.24.4"
+)
+
+var (
+	collectorCRName = flag.String("collector-cr-name", "collector-cr", "name of the collector cr to read from and write to")
+	cacheDir        = flag.String("cache-dir", "/var/lib/trivy", "path to the cache dir")
+	severity        = flag.String("severity", "CRITICAL", "list of severity levels to report")
+	ignoreUnfixed   = flag.Bool("ignore-unfixed", true, "report only fixed vulnerabilities")
+	vulnTypes       = flag.String("vuln-type", "os,library", "comma separated list of vulnerability types")
+	securityChecks  = flag.String("security-checks", "vuln,secret", "comma-separated list of what security issues to detect")
+
+	// Will be modified by parseCommaSeparatedOptions() to reflect the `severity` CLI flag
+	// These are the only recognized severities and the keys of this map should never be modified.
+	severityMap = map[string]bool{
+		severityCritical: false,
+		severityHigh:     false,
+		severityMedium:   false,
+		severityLow:      false,
+		severityUnknown:  false,
+	}
+
+	// Will be modified by parseCommaSeparatedOptions() to reflect the `security-checks` CLI flag
+	// These are the only recognized security checks and the keys of this map should never be modified.
+	securityCheckMap = map[string]bool{
+		securityCheckVuln:   false,
+		securityCheckSecret: false,
+		securityCheckConfig: false,
+	}
+
+	// Will be modified by parseCommaSeparatedOptions()  to reflect the `vuln-type` CLI flag
+	// These are the only recognized vulnerability types and the keys of this map should never be modified.
+	vulnTypeMap = map[string]bool{
+		vulnTypeOs:      false,
+		vulnTypeLibrary: false,
+	}
+
+	log = logf.Log.WithName("eraser")
```

Comment:
do we want to say "scanner" here? and perhaps `WithValues("provider", "trivy")`?